### ISSUE TITLE
Array over size limit

### ISSUE THREAD

**Issue by moadben** [pachyderm/pachyderm#1004]
Hi everyone,

It seems that after a certain point, attempting to list files in a commit will return this error:

```
invalid header field value "gorethink: Array over size limit `100000`. in: \n[{Right=229, Branch=\"master\", Left=0}].ConcatMap(func(var_4242481 r.Term) r.Term { return r.DB(\"pachyderm_pfs\").Table(\"Diffs\").OrderBy(index=\"DiffPrefixIndex\").Between([\"prospector_input\", \"/\", [var_4242481.Field(\"Branch\"), var_4242481.Field(\"Left\")]], [\"prospector_input\", \"/\", [var_4242481.Field(\"Branch\"), var_4242481.Field(\"Right\")]], left_bound=\"closed\", right_bound=\"closed\") }).Group(\"Path\").Ungroup().Field(\"reduction\").Map(func(var_4242482 r.Term) r.Term { return var_4242482.Fold({Size=0, FileType=0, BlockRefs=[], Modified=<nil>, Clock=<nil>, ID=\"\", Repo=\"\", Path=\"\", Delete=false}, func(var_4242483, var_4242484 r.Term) r.Term { return r.Branch(var_4242483.Field(\"FileType\").Ne(0).And(var_4242484.Field(\"FileType\").Ne(0).And(var_4242483.Field(\"FileType\").Ne(var_4242484.Field(\"FileType\")))), r.Error(\"file type conflict\"), r.Branch(var_4242484.Field(\"Delete\"), var_4242483.Merge(var_4242484).Merge({Delete=var_4242483.Field(\"Delete\").Or(var_4242484.Field(\"Delete\"))}), var_4242483.Merge(var_4242484).Merge({BlockRefs=var_4242483.Field(\"BlockRefs\").Add(var_4242484.Field(\"BlockRefs\")), Size=var_4242483.Field(\"Size\").Add(var_4242484.Field(\"Size\")), Delete=var_4242483.Field(\"Delete\").Or(var_4242484.Field(\"Delete\"))}))) }) }).Filter(func(var_4242485 r.Term) r.Term { return var_4242485.Field(\"FileType\").Ne(0) }).Group(func(var_4242486 r.Term) r.Term { return r.Branch(\"/\".Eq(\"/\"), var_4242486.Field(\"Path\").Split(\"/\").Nth(1), var_4242486.Field(\"Path\").Split(\"/\", 1).Nth(1).Split(\"/\").Nth(1)) }).Reduce(func(var_4242487, var_4242488 r.Term) r.Term { return r.Branch(var_4242487.Field(\"Path\").Lt(var_4242488.Field(\"Path\")), var_4242487.Merge({Size=var_4242487.Field(\"Size\").Add(var_4242488.Field(\"Size\"))}), var_4242488.Merge({Size=var_4242487.Field(\"Size\").Add(var_4242488.Field(\"Size\"))})) }).Ungroup().Field(\"reduction\").OrderBy(\"Path\")"
```

The first time this popped up for me was on my 229th commit with about ~30,000 files total in the repo, and attempting to list the files in any commit afterwards returns the same error.

Thanks,
Moad.

**Comment by jdoliner** [comment]
Hi @moadben, really sorry you hit this.
In the short term we can get this fixed for you guys by upping the array limit.
I'll make that change and do a point release for you guys so you can redeploy and not have this error.

That of course will only take us so far. In the long term we should switch to doing `list-file` in a streaming fashion so it won't hit the array limit. That should also be more efficient.

**Comment by moadben** [comment]
@jdoliner cool, thank you very much!

**Comment by jdoliner** [comment]
A short term fix for this has landed in master and will be released in 1.2.3. For further work on this issue track #1006.
